nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/AbstractMongoProcessor.java

```diff
@@ -95,15 +100,36 @@
 
     public static final PropertyDescriptor WRITE_CONCERN = new PropertyDescriptor.Builder()
             .name("Write Concern")
+            .displayName("Write Concern")
             .description("The write concern to use")
             .required(true)
             .allowableValues(WRITE_CONCERN_ACKNOWLEDGED, WRITE_CONCERN_UNACKNOWLEDGED, WRITE_CONCERN_FSYNCED, WRITE_CONCERN_JOURNALED,
                     WRITE_CONCERN_REPLICA_ACKNOWLEDGED, WRITE_CONCERN_MAJORITY)
             .defaultValue(WRITE_CONCERN_ACKNOWLEDGED)
             .build();
 
+    static final PropertyDescriptor RESULTS_PER_FLOWFILE = new PropertyDescriptor.Builder()
+            .name("results-per-flowfile")
+            .displayName("Results Per FlowFile")
+            .description("How many results to put into a flowfile at once. The whole body will be treated as a JSON array of results.")
+            .required(false)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("1")
+            .build();
+
+    static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size")
+            .displayName("Batch Size")
+            .description("The number of elements returned from the server in one batch")
+            .required(false)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("1")
+            .build();
+
     static List<PropertyDescriptor> descriptors = new ArrayList<>();
 
+    static final Relationship REL_SUCCESS = new Relationship.Builder().name("success").description("All files are routed to success").build();
```

Comment:
You may not want this in the Abstract class as the description may change based on who is using it (see my other comments). In this case, the stock text description is common to many Get processors because if any flow files are produced, they are routed to success. In RunMongoAggregation, if there is an incoming flow file, then it is not routed to success; rather the results of a successful aggregation are written to success, and the original is written to "original", and on failure, the incoming flow file should be transferred to failure.